### PR TITLE
pekko-http-jackson: upgrade jackson to 2.16

### PR DESCRIPTION
**Pull request by pjfanning**
add config support for setting read and write constraints

### CODE REVIEW

**Comment by mkurz** [comment]
For which release do you plan to merge this? 1.1 or 2?

**Comment by pjfanning** [comment]
@mkurz I think 1.1 is ok for this. Users can also use https://mvnrepository.com/artifact/com.github.pjfanning/pekko-http-jackson if they need support sooner

**Comment by mkurz** [comment]
It's not about getting it sooner. We want to stay with Jackson 2.14 for a while (until new major Play release next year). After 1.1 got released, how long will you patch 1.0?

**Comment by pjfanning** [comment]
I guess the aim would be to fix major bugs on the 1.0.x for at least a year after a 1.1 release. If and when we do a Pekko HTTP 1.1 release, I am not expecting it to be a release that we will be railroading users onto.

**Comment by pjfanning** [comment]
@mkurz does Play even use the org.apache.pekko:pekko-http-jackson jar? I don't see a dependency on it in https://github.com/playframework/playframework.

Jackson 2.15 and 2.16 have some important fixes, including security hardening and I don't want to at least facilitate users to benefit from these fixes.

File: http-marshallers-java/http-jackson/src/main/resources/reference.conf [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# SPDX-License-Identifier: Apache-2.0
+
+##################################
+# Pekko HTTP Jackson Config File #
+##################################
+
+# This is the reference config file that contains all the default settings.
+# Make your edits/overrides in your application.conf.
+
+pekko.http.jackson {
```
Comment (He-Pin) [inline http-marshallers-java/http-jackson/src/main/resources/reference.conf:10]: pekko.http.marsharllers.jackson?

File: http-marshallers-java/http-jackson/src/main/resources/reference.conf [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  read {
+    # see https://www.javadoc.io/static/com.fasterxml.jackson.core/jackson-core/2.16.1/com/fasterxml/jackson/core/StreamReadConstraints.html
+    # these defaults are the same as the defaults in `StreamReadConstraints`
+    max-nesting-depth = 1000
+    max-number-length = 1000
+    max-string-length = 20000000
+    max-name-length = 50000
+    # max-document-length of -1 means unlimited
+    max-document-length = -1
+  }
```
Comment (He-Pin) [inline http-marshallers-java/http-jackson/src/main/resources/reference.conf:20]: Add a new blank line

File: http-marshallers-java/http-jackson/src/main/resources/reference.conf (hunk 1 above)
Comment (He-Pin) [reply http-marshallers-java/http-jackson/src/main/resources/reference.conf:10]: Or 
pekko.http.serialization.jackson {

File: http-marshallers-java/http-jackson/src/main/resources/reference.conf (hunk 2 above)
Comment (pjfanning) [reply http-marshallers-java/http-jackson/src/main/resources/reference.conf:20]: done

File: http-marshallers-java/http-jackson/src/main/resources/reference.conf (hunk 1 above)
Comment (pjfanning) [reply http-marshallers-java/http-jackson/src/main/resources/reference.conf]: renamed as pekko.http.marsharllers.jackson

**Review by He-Pin — APPROVED** [review]
lgtm

**Review by jxnu-liguobin — APPROVED** [review]
lgtm

**Pull request merged**